bin/asc.js

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env node
+
+import childProcess from "child_process";
+
+const [ nodePath, thisPath, ...args ] = process.argv;
+const nodeArgs = process.execArgv;
+
+const hasSourceMaps = nodeArgs.includes("--enable-source-maps");
+const hasCustomArgs = args.includes("--");
+
+if (!hasSourceMaps || hasCustomArgs) {
+  if (!hasSourceMaps) {
+    nodeArgs.push("--enable-source-maps");
+  }
+  if (hasCustomArgs) {
+    const index = args.indexOf("--");
```

Comment:
Btw could you simplify this as well and reuse `args.indexOf("--")` for `hasCustomArgs`?